connectors/src/connectors/google_drive/temporal/workflows.ts

```diff
@@ -68,6 +72,25 @@ export async function googleDriveFullSync({
     foldersToBrowse = await getFoldersToSync(connectorId);
   }
 
+  setHandler(folderUpdatesSignal, (folderUpdates: FolderUpdatesSignal[]) => {
+    // If we get a signal, update the workflow state by adding/removing folder ids.
+    for (const { action, folderId } of folderUpdates) {
+      switch (action) {
+        case "added":
+          foldersToBrowse!.push(folderId);
+          break;
+        case "removed":
+          foldersToBrowse!.splice(foldersToBrowse!.indexOf(folderId), 1);
+          break;
+        default:
+          assertNever(
```

Comment:
We have our own assert never that we typically use

Reply:
You mean the one from dust-types ? I used this initially but it break the worker (error at startup) as it will pull tons of code that contains forbidden modules (because they are non deterministic).

The one in temporal doesn't create this problem (talked about it with @flvndvd before using it).

Reply:
The problem with using `assertNever` (the one from `types`) in Temporal workflow code is that it leads to bundling the entire types package, which imports native APIS that aren't supported by the Node.js environment of Temporal workflows.

Reply:
Indeed, didn't realize this was temporal workflow code 👍  fair to not import from types